src/prefect/runner/runner.py

```diff
@@ -570,17 +582,29 @@ async def execute_flow_run(
                     self._submitting_flow_run_ids.add(flow_run_id)
                     flow_run = await self._client.read_flow_run(flow_run_id)
 
-                    pid = await self._runs_task_group.start(
+                    process: anyio.abc.Process = await self._runs_task_group.start(
                         partial(
                             self._submit_run_and_capture_errors,
                             flow_run=flow_run,
                             entrypoint=entrypoint,
+                            command=command,
+                            cwd=cwd,
+                            env=env,
+                            stream_output=stream_output,
                         ),
                     )
+                    if task_status:
+                        task_status.started(process.pid)
 
-                    self._flow_run_process_map[flow_run.id] = ProcessMapEntry(
-                        pid=pid, flow_run=flow_run
-                    )
+                    if TYPE_CHECKING:
+                        assert self._flow_run_process_map_lock is not None
+
+                    async with self._flow_run_process_map_lock:
+                        # Only add the process to the map if it is still running
+                        if process.returncode is None:
+                            self._flow_run_process_map[flow_run.id] = ProcessMapEntry(
+                                pid=process.pid, flow_run=flow_run
+                            )
```

Comment:
I added this lock and check because I discovered that a race condition would cause the runner would hang if the process exited before reaching this point.